src/v/kafka/server/handlers/produce.cc

```diff
@@ -234,6 +234,77 @@ ss::future<produce_response::partition> finalize_request_with_error_code(
         .partition_index = ntp.tp.partition, .error_code = ec});
 }
 
+/**
+ * @brief Validate the timestamps of the batch, they have to be within a window
+ * to the broker's time. returns the new timestamp to set as max_timestamp to
+ * the batch, if present
+ */
+static auto validate_batch_timestamps(
+  model::ntp const& ntp,
+  model::record_batch const& batch,
+  model::timestamp_type timestamp_type,
```

Comment:
I think we only need to pass in the `batch.header()` field and can then remove `batch` and `timestamp_type`.

Reply:
i have to check but timestamp_type needs to be computed from cluster settings and setting a timestamp_time would force a CRC operation

Reply:
Ah right so maybe just batch -> header?